compiler/src/dotty/tools/dotc/core/tasty/Attributes.scala

```diff
@@ -3,4 +3,12 @@ package dotty.tools.dotc.core.tasty
 class Attributes(
   val scala2StandardLibrary: Boolean,
   val explicitNulls: Boolean,
-)
+  val isJava: Boolean,
+  val isOutline: Boolean,
+) {
+  def nonEmpty: Boolean = scala2StandardLibrary || explicitNulls || isJava || isOutline
+}
+
+object Attributes {
+  lazy val empty = new Attributes(false, false, false, false)
+}
```

Comment:
You should base this on top of e8428e18ae7e0f4e4b8caee56569a4c805d541ce. This would avoid conflicts with other PRs and probably simplify a bit your changes on the attributes.

Reply:
I have rebased on #19092